File: app/src/main/java/com/example/rooster/veterinary/prescriptions/PrescriptionManagementScreen.kt
package com.example.rooster.veterinary.prescriptions

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.lifecycle.ViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.update
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import kotlin.random.Random

// --- Data Classes ---
data class MedicationCatalogItem(
    val name: String,
    val unit: String,
    val requiresVetAuth: Boolean
)

data class Prescription(
    val prescriptionId: String,
    val vetId: String,
    val vetName: String,
    val patientId: String,
    val patientName: String,
    val userId: String, // Owner
    val dateIssued: Date,
    val medicationName: String,
    var dosage: String,
    var frequency: String,
    var duration: String,
    var quantityDispensed: String? = null,
    var dispensingPharmacyId: String? = null,
    var instructions: String,
    var refillsAllowed: Int,
    var refillsRemaining: Int,
    var status: String, // issued, dispensed, cancelled, expired, recommended
    var notes: String?,
    var dateDispensed: Date? = null,
    var lastUpdatedBy: String? = null,
    var lastUpdatedAt: Date? = null
) {
    fun getFormattedDate(date: Date?): String =
        date?.let { SimpleDateFormat("yyyy-MM-dd HH:mm", Locale.getDefault()).format(it) } ?: "N/A"
}

// --- ViewModel ---
class PrescriptionViewModel : ViewModel() {
    private val _prescriptions = MutableStateFlow<List<Prescription>>(emptyList())
    val prescriptions: StateFlow<List<Prescription>> = _prescriptions

    val medicationCatalog = listOf(
        MedicationCatalogItem("Amoxicillin 250mg tablets", "tablet", true),
        MedicationCatalogItem("Meloxicam 1mg/ml Oral Suspension", "ml", true),
        MedicationCatalogItem("Ivermectin Pour-On", "ml", true),
        MedicationCatalogItem("Saline Eye Wash", "bottle", false),
        MedicationCatalogItem("Medicated Shampoo", "bottle", false),
        MedicationCatalogItem("Fenbendazole Granules 22.2%", "gram", true)
    )
    val knownVetIds = (1..3).map { "vet${it.toString().padStart(3,'0')}" }
    val knownPatientIds = (1..5).map { "pet${it.toString().padStart(3,'0')}" }
    val knownUserIds = (1..5).map { "farmer${it.toString().padStart(3,'0')}" }
    val prescriptionStatuses = listOf("issued", "dispensed", "cancelled", "expired", "recommended")


    init {
        loadMockPrescriptions()
    }

    private fun loadMockPrescriptions(count: Int = 5) {
        val now = Date()
        val tempPrescriptions = mutableListOf<Prescription>()
        for (i in 0 until count) {
            val medInfo = medicationCatalog.random()
            val issueDate = Date(now.time - Random.nextLong(0, 90 * 86400000L))
            val status = if (medInfo.requiresVetAuth) listOf("issued", "dispensed", "cancelled").random() else listOf("recommended", "dispensed").random()
            val p = Prescription(
                prescriptionId = "rx_${issueDate.time}_$i",
                vetId = knownVetIds.random(),
                vetName = "Dr. Mock Vet", // Simplified
                patientId = knownPatientIds.random(),
                patientName = "Mock Patient", // Simplified
                userId = knownUserIds.random(),
                dateIssued = issueDate,
                medicationName = medInfo.name,
                dosage = "${Random.nextInt(1, 3)} ${medInfo.unit}(s)",
                frequency = listOf("Once daily", "Twice daily", "As needed").random(),
                duration = if (medInfo.requiresVetAuth) "${Random.nextInt(3,14)} days" else "Until symptoms resolve",
                instructions = "Follow directions carefully.",
                refillsAllowed = if (medInfo.requiresVetAuth) Random.nextInt(0, 3) else 0,
                refillsRemaining = 0, // Will be set
                status = status,
                notes = if (medInfo.requiresVetAuth) "Ensure full course." else "OTC recommendation."
            )
            p.refillsRemaining = p.refillsAllowed
            if (status == "dispensed") {
                p.quantityDispensed = "${Random.nextInt(10,50)} ${medInfo.unit}(s)"
                p.dispensingPharmacyId = "pharm${Random.nextInt(1,3)}"
                p.dateDispensed = Date(issueDate.time + Random.nextLong(0, 2 * 86400000L))
                if(p.refillsRemaining > 0) p.refillsRemaining--
            }
            tempPrescriptions.add(p)
        }
        _prescriptions.value = tempPrescriptions.sortedByDescending { it.dateIssued }
    }

    fun createPrescription(vetId: String, patientId: String, userId: String, medName: String, dosage: String, freq: String, duration: String, instr: String, refills: Int, notes: String?) {
        val medInfo = medicationCatalog.find { it.name == medName } ?: return // Should not happen if UI uses catalog
        val newRx = Prescription(
            prescriptionId = "rx_${Date().time}_${Random.nextInt(1000)}",
            vetId = vetId, vetName = "Dr. Current User", patientId = patientId, patientName = "Patient $patientId", userId = userId,
            dateIssued = Date(), medicationName = medName, dosage = dosage, frequency = freq, duration = duration,
            instructions = instr, refillsAllowed = refills, refillsRemaining = refills, status = "issued", notes = notes
        )
        _prescriptions.update { (listOf(newRx) + it).sortedByDescending { rx -> rx.dateIssued } }
    }

    fun updatePrescriptionStatus(rxId: String, newStatus: String, updatedBy: String, qtyDispensed: String? = null, pharmacyId: String? = null) {
        _prescriptions.update { list ->
            list.map {
                if (it.prescriptionId == rxId) {
                    it.copy(
                        status = newStatus,
                        lastUpdatedAt = Date(),
                        lastUpdatedBy = updatedBy,
                        quantityDispensed = if (newStatus == "dispensed") qtyDispensed ?: it.quantityDispensed else it.quantityDispensed,
                        dispensingPharmacyId = if (newStatus == "dispensed") pharmacyId ?: it.dispensingPharmacyId else it.dispensingPharmacyId,
                        dateDispensed = if (newStatus == "dispensed") Date() else it.dateDispensed,
                        refillsRemaining = if (newStatus == "dispensed" && it.refillsRemaining > 0) it.refillsRemaining -1 else it.refillsRemaining
                    )
                } else it
            }
        }
    }
}


// --- Composable Screen ---
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun PrescriptionManagementScreen(viewModel: PrescriptionViewModel = androidx.lifecycle.viewmodel.compose.viewModel()) {
    val prescriptions by viewModel.prescriptions.collectAsState()
    var showCreateDialog by remember { mutableStateOf(false) }
    var showUpdateStatusDialog by remember { mutableStateOf(false) }
    var selectedPrescription by remember { mutableStateOf<Prescription?>(null) }

    if (showCreateDialog) {
        CreatePrescriptionDialog(viewModel = viewModel, onDismiss = { showCreateDialog = false })
    }
    if (showUpdateStatusDialog && selectedPrescription != null) {
        UpdatePrescriptionStatusDialog(
            prescription = selectedPrescription!!,
            viewModel = viewModel,
            onDismiss = { showUpdateStatusDialog = false; selectedPrescription = null }
        )
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Prescription Management") },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = Color(0xFF26A69A)) // Teal
            )
        },
        floatingActionButton = {
            FloatingActionButton(onClick = { showCreateDialog = true }) {
                Icon(Icons.Filled.Add, "Create New Prescription")
            }
        }
    ) { paddingValues ->
        LazyColumn(modifier = Modifier.padding(paddingValues).padding(16.dp)) {
            if (prescriptions.isEmpty()) {
                item { Text("No prescriptions found.") }
            } else {
                items(prescriptions, key = { it.prescriptionId }) { rx ->
                    PrescriptionCard(rx) {
                        selectedPrescription = rx
                        showUpdateStatusDialog = true
                    }
                }
            }
        }
    }
}

@Composable
fun PrescriptionCard(rx: Prescription, onCardClick: () -> Unit) {
    Card(modifier = Modifier.fillMaxWidth().padding(vertical = 6.dp).clickable(onClick = onCardClick), elevation = CardDefaults.cardElevation(2.dp)) {
        Column(Modifier.padding(12.dp), verticalArrangement = Arrangement.spacedBy(4.dp)) {
            Text("${rx.medicationName} (ID: ${rx.prescriptionId.takeLast(6)})", style = MaterialTheme.typography.titleMedium)
            Text("Patient: ${rx.patientName} (ID: ${rx.patientId}) - Owner: ${rx.userId}")
            Text("Vet: ${rx.vetName} (ID: ${rx.vetId})")
            Text("Issued: ${rx.getFormattedDate(rx.dateIssued)}")
            Text("Dosage: ${rx.dosage}, ${rx.frequency} for ${rx.duration}")
            Text("Instructions: ${rx.instructions}")
            Text("Refills: ${rx.refillsRemaining}/${rx.refillsAllowed}")
            Text("Status: ${rx.status}", color = when(rx.status){
                "issued" -> Color.Blue
                "dispensed" -> Color.Green
                "cancelled", "expired" -> Color.Red
                else -> Color.DarkGray
            })
            rx.notes?.let { Text("Notes: $it") }
            if (rx.status == "dispensed") {
                Text("Dispensed: ${rx.quantityDispensed ?: ""} by ${rx.dispensingPharmacyId ?: ""} on ${rx.getFormattedDate(rx.dateDispensed)}")
            }
            rx.lastUpdatedAt?.let { Text("Last Update: ${rx.getFormattedDate(it)} by ${rx.lastUpdatedBy ?: ""}", style = MaterialTheme.typography.bodySmall) }
        }
    }
}

@Composable
fun CreatePrescriptionDialog(viewModel: PrescriptionViewModel, onDismiss: () -> Unit) {
    // Basic fields for simplicity, can be expanded
    var patientId by remember { mutableStateOf(viewModel.knownPatientIds.first()) }
    var medicationName by remember { mutableStateOf(viewModel.medicationCatalog.first().name) }
    var dosage by remember { mutableStateOf("") }
    var frequency by remember { mutableStateOf("") }
    var duration by remember { mutableStateOf("") }
    var instructions by remember { mutableStateOf("") }
    var refills by remember { mutableStateOf("0") }
    var notes by remember { mutableStateOf("") }

    AlertDialog(
        onDismissRequest = onDismiss,
        title = { Text("Create New Prescription") },
        text = {
            LazyColumn(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                item { ExposedDropdownMenuForOptions(label = "Patient ID", options = viewModel.knownPatientIds, selectedOption = patientId, onOptionSelected = { patientId = it }) }
                item { ExposedDropdownMenuForOptions(label = "Medication", options = viewModel.medicationCatalog.map { it.name }, selectedOption = medicationName, onOptionSelected = { medicationName = it }) }
                item { OutlinedTextField(value = dosage, onValueChange = { dosage = it }, label = { Text("Dosage (e.g., 1 tablet)") }) }
                item { OutlinedTextField(value = frequency, onValueChange = { frequency = it }, label = { Text("Frequency (e.g., Twice daily)") }) }
                item { OutlinedTextField(value = duration, onValueChange = { duration = it }, label = { Text("Duration (e.g., 7 days)") }) }
                item { OutlinedTextField(value = instructions, onValueChange = { instructions = it }, label = { Text("Instructions") }) }
                item { OutlinedTextField(value = refills, onValueChange = { refills = it.filter(Char::isDigit) }, label = { Text("Refills Allowed") }, keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number)) }
                item { OutlinedTextField(value = notes, onValueChange = { notes = it }, label = { Text("Notes (Optional)") }) }
            }
        },
        confirmButton = {
            Button(onClick = {
                viewModel.createPrescription(
                    vetId = "vet_current_user", // Placeholder
                    patientId = patientId,
                    userId = "farmer_current_user", // Placeholder
                    medName = medicationName,
                    dosage = dosage, freq = frequency, duration = duration, instr = instructions,
                    refills = refills.toIntOrNull() ?: 0, notes = notes.ifBlank { null }
                )
                onDismiss()
            }) { Text("Create") }
        },
        dismissButton = { Button(onClick = onDismiss) { Text("Cancel") } }
    )
}

@Composable
fun UpdatePrescriptionStatusDialog(prescription: Prescription, viewModel: PrescriptionViewModel, onDismiss: () -> Unit) {
    var newStatus by remember { mutableStateOf(prescription.status) }
    var qtyDispensed by remember { mutableStateOf(prescription.quantityDispensed ?: "") }
    var pharmacyId by remember { mutableStateOf(prescription.dispensingPharmacyId ?: "") }

    AlertDialog(
        onDismissRequest = onDismiss,
        title = { Text("Update Rx: ${prescription.prescriptionId.takeLast(6)}") },
        text = {
            Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                Text("Medication: ${prescription.medicationName}")
                ExposedDropdownMenuForOptions(label = "New Status", options = viewModel.prescriptionStatuses, selectedOption = newStatus, onOptionSelected = {newStatus = it})
                if (newStatus == "dispensed") {
                    OutlinedTextField(value = qtyDispensed, onValueChange = {qtyDispensed = it}, label = {Text("Quantity Dispensed")})
                    OutlinedTextField(value = pharmacyId, onValueChange = {pharmacyId = it}, label = {Text("Dispensing Pharmacy ID")})
                }
            }
        },
        confirmButton = { Button(onClick = {
            viewModel.updatePrescriptionStatus(prescription.prescriptionId, newStatus, "admin_compose", if(newStatus == "dispensed") qtyDispensed else null, if(newStatus == "dispensed") pharmacyId else null)
            onDismiss()
        }) {Text("Update Status")} },
        dismissButton = { Button(onClick = onDismiss) {Text("Cancel")} }
    )
}

// Re-using ExposedDropdownMenuForOptions, assuming it's accessible
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ExposedDropdownMenuForOptions( // Copied for standalone preview if needed
    label: String,
    options: List<String>,
    selectedOption: String,
    onOptionSelected: (String) -> Unit
) {
    var expanded by remember { mutableStateOf(false) }
    ExposedDropdownMenuBox(
        expanded = expanded,
        onExpandedChange = { expanded = !expanded }
    ) {
        OutlinedTextField(
            value = selectedOption,
            onValueChange = {}, readOnly = true,
            label = { Text(label) },
            trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded = expanded) },
            modifier = Modifier.menuAnchor().fillMaxWidth()
        )
        ExposedDropdownMenu(
            expanded = expanded,
            onDismissRequest = { expanded = false }
        ) {
            options.forEach { option ->
                DropdownMenuItem(
                    text = { Text(option) },
                    onClick = {
                        onOptionSelected(option)
                        expanded = false
                    }
                )
            }
        }
    }
}

@Preview(showBackground = true, heightDp = 800)
@Composable
fun PreviewPrescriptionManagementScreen() {
    MaterialTheme {
        PrescriptionManagementScreen(viewModel = PrescriptionViewModel())
    }
}

File: app/veterinary/prescription_management_screen.py
import datetime
import random

class PrescriptionManagementScreen:
    def __init__(self):
        self.prescriptions = [] # List of all prescriptions
        # Mock data - In a real system, these would come from other modules/DB
        self.known_vet_ids = [f"vet{str(i).zfill(3)}" for i in range(1, 4)]
        self.known_vet_names = {"vet001": "Dr. Alice Smith", "vet002": "Dr. Bob Johnson", "vet003": "Dr. Carol White"}
        self.known_patient_ids = [f"pet{str(i).zfill(3)}" for i in range(1, 6)]
        self.known_patient_names = {"pet001": "Bessie", "pet002": "Charlie", "pet003": "Daisy", "pet004": "Rocky", "pet005": "Whiskers"}
        self.known_user_ids = [f"farmer{str(i).zfill(3)}" for i in range(1, 6)] # Owners

        self.medications_catalog = [ # Simplified catalog
            {"name": "Amoxicillin 250mg tablets", "unit": "tablet", "requires_vet_auth": True},
            {"name": "Meloxicam 1mg/ml Oral Suspension", "unit": "ml", "requires_vet_auth": True},
            {"name": "Ivermectin Pour-On", "unit": "ml", "requires_vet_auth": True},
            {"name": "Saline Eye Wash", "unit": "bottle", "requires_vet_auth": False},
            {"name": "Medicated Shampoo", "unit": "bottle", "requires_vet_auth": False},
            {"name": "Fenbendazole Granules 22.2%", "unit": "gram", "requires_vet_auth": True}
        ]
        self._initialize_mock_prescriptions()

    def _initialize_mock_prescriptions(self, count=5):
        for i in range(count):
            vet_id = random.choice(self.known_vet_ids)
            patient_id = random.choice(self.known_patient_ids)
            user_id = f"farmer{str(random.randint(1,5)).zfill(3)}" # Mock owner
            med_info = random.choice(self.medications_catalog)

            # Ensure only authorized meds are prescribed if vet auth is required
            if med_info["requires_vet_auth"]:
                status = random.choice(["issued", "dispensed", "cancelled"])
            else: # OTC meds might not always be "prescribed" but could be "recommended"
                status = random.choice(["recommended", "dispensed"])

            prescription_date = datetime.datetime.now() - datetime.timedelta(days=random.randint(0, 90))

            new_prescription = {
                "prescription_id": f"rx_{prescription_date.strftime('%Y%m%d')}_{random.randint(1000,9999)}",
                "vet_id": vet_id,
                "vet_name": self.known_vet_names.get(vet_id, "Unknown Vet"),
                "patient_id": patient_id,
                "patient_name": self.known_patient_names.get(patient_id, "Unknown Patient"),
                "user_id": user_id, # Owner
                "date_issued": prescription_date.isoformat(),
                "medication_name": med_info["name"],
                "dosage": f"{random.randint(1,3)} {med_info['unit']}(s)",
                "frequency": random.choice(["Once daily", "Twice daily", "As needed", "Every 8 hours"]),
                "duration": f"{random.randint(3, 14)} days" if med_info["requires_vet_auth"] else "Until symptoms resolve",
                "quantity_dispensed": None, # To be filled if status is 'dispensed'
                "dispensing_pharmacy_id": None, # If applicable
                "instructions": random.choice(["Administer with food.", "Keep refrigerated.", "Shake well before use.", "For external use only."]),
                "refills_allowed": random.randint(0,2) if med_info["requires_vet_auth"] else 0,
                "refills_remaining": 0, # Will be set based on allowed
                "status": status, # issued, dispensed, cancelled, expired, recommended
                "notes": "Ensure full course is completed." if med_info["requires_vet_auth"] else "Follow label directions."
            }
            new_prescription["refills_remaining"] = new_prescription["refills_allowed"]
            if status == "dispensed":
                new_prescription["quantity_dispensed"] = f"{random.randint(10,50)} {med_info['unit']}(s)" # Mock quantity
                new_prescription["dispensing_pharmacy_id"] = f"pharm{random.randint(1,3)}"
                new_prescription["date_dispensed"] = (prescription_date + datetime.timedelta(days=random.randint(0,2))).isoformat()

            self.prescriptions.append(new_prescription)
        self.prescriptions.sort(key=lambda x: x["date_issued"], reverse=True)

    def create_prescription(self, vet_id, patient_id, user_id, medication_name, dosage, frequency, duration, instructions, refills_allowed=0, notes=""):
        if vet_id not in self.known_vet_ids:
            print(f"Error: Veterinarian {vet_id} not authorized or found.")
            return None

        med_info = next((med for med in self.medications_catalog if med["name"] == medication_name), None)
        if not med_info:
            print(f"Error: Medication '{medication_name}' not found in catalog.")
            return None
        if med_info["requires_vet_auth"] and vet_id not in self.known_vet_ids : # Redundant check here, but good for logic
             print(f"Error: Veterinarian {vet_id} not authorized to prescribe {medication_name}.")
             return None

        prescription_date = datetime.datetime.now()
        new_rx = {
            "prescription_id": f"rx_{prescription_date.strftime('%Y%m%d%H%M%S')}_{random.randint(100,999)}",
            "vet_id": vet_id,
            "vet_name": self.known_vet_names.get(vet_id, "Unknown Vet"),
            "patient_id": patient_id,
            "patient_name": self.known_patient_names.get(patient_id, "Unknown Patient"),
            "user_id": user_id,
            "date_issued": prescription_date.isoformat(),
            "medication_name": medication_name,
            "dosage": dosage,
            "frequency": frequency,
            "duration": duration,
            "instructions": instructions,
            "refills_allowed": refills_allowed,
            "refills_remaining": refills_allowed,
            "status": "issued",
            "notes": notes,
            "quantity_dispensed": None,
            "dispensing_pharmacy_id": None
        }
        self.prescriptions.insert(0, new_rx) # Add to top of the list
        print(f"Prescription {new_rx['prescription_id']} created for {medication_name} for patient {patient_id} by {vet_id}.")
        return new_rx["prescription_id"]

    def get_prescription_details(self, prescription_id):
        for rx in self.prescriptions:
            if rx["prescription_id"] == prescription_id:
                return rx
        return None

    def get_prescriptions_for_patient(self, patient_id, status_filter=None):
        results = [rx for rx in self.prescriptions if rx["patient_id"] == patient_id]
        if status_filter:
            results = [rx for rx in results if rx["status"] == status_filter]
        return results

    def get_prescriptions_by_vet(self, vet_id, status_filter=None):
        results = [rx for rx in self.prescriptions if rx["vet_id"] == vet_id]
        if status_filter:
            results = [rx for rx in results if rx["status"] == status_filter]
        return results

    def update_prescription_status(self, prescription_id, new_status, admin_id_or_pharmacy_id, quantity_dispensed=None):
        rx = self.get_prescription_details(prescription_id)
        if not rx:
            print(f"Error: Prescription {prescription_id} not found.")
            return False

        valid_statuses = ["issued", "dispensed", "cancelled", "expired", "recommended"]
        if new_status not in valid_statuses:
            print(f"Error: Invalid status '{new_status}'.")
            return False

        rx["status"] = new_status
        rx["last_updated_by"] = admin_id_or_pharmacy_id
        rx["last_updated_at"] = datetime.datetime.now().isoformat()

        if new_status == "dispensed":
            if not quantity_dispensed:
                print("Error: Quantity dispensed must be provided for 'dispensed' status.")
                # Revert status if not provided, or handle as partial dispense
                # For now, just warn. A real system would be stricter.
            rx["quantity_dispensed"] = quantity_dispensed
            rx["dispensing_pharmacy_id"] = admin_id_or_pharmacy_id # Assuming pharmacy ID is passed
            rx["date_dispensed"] = datetime.datetime.now().isoformat()
            if rx["refills_remaining"] > 0 :
                rx["refills_remaining"] -=1 # This logic might be more complex (e.g. full vs partial)

        print(f"Prescription {prescription_id} status updated to '{new_status}' by {admin_id_or_pharmacy_id}.")
        return True

    def display_screen(self):
        print("---- Prescription Management Screen ----")

        print("\n-- Create New Prescription (Example) --")
        # Select a vet, patient, user for the new prescription
        demo_vet_id = self.known_vet_ids[0]
        demo_patient_id = self.known_patient_ids[0]
        demo_user_id = self.known_user_ids[0]
        demo_med_info = self.medications_catalog[0] # Amoxicillin

        new_rx_id = self.create_prescription(
            vet_id=demo_vet_id,
            patient_id=demo_patient_id,
            user_id=demo_user_id,
            medication_name=demo_med_info["name"],
            dosage=f"1 {demo_med_info['unit']}",
            frequency="Twice daily",
            duration="7 days",
            instructions="Administer with a small amount of food.",
            refills_allowed=1,
            notes="Re-evaluate if no improvement in 3 days."
        )
        if new_rx_id:
            print(f"  Successfully created prescription: {new_rx_id}")

        print("\n-- Prescriptions for Patient (e.g., pet001 - Bessie, Max 2) --")
        patient_rxs = self.get_prescriptions_for_patient(self.known_patient_ids[0])[:2]
        if not patient_rxs:
            print(f"  No prescriptions found for patient {self.known_patient_ids[0]}.")
        else:
            for rx in patient_rxs:
                print(f"  ID: {rx['prescription_id']}, Med: {rx['medication_name']}, Date: {rx['date_issued'][:10]}, Status: {rx['status']}")
                print(f"    Dosage: {rx['dosage']} {rx['frequency']} for {rx['duration']}. Refills Left: {rx['refills_remaining']}")

        print("\n-- Update Prescription Status (Example) --")
        if new_rx_id: # Use the newly created one if available
            target_rx_id_for_update = new_rx_id
        elif self.prescriptions: # Else pick first available
             target_rx_id_for_update = self.prescriptions[0]["prescription_id"]
        else:
            target_rx_id_for_update = None

        if target_rx_id_for_update:
            print(f"  Attempting to mark prescription {target_rx_id_for_update} as 'dispensed'.")
            # Find the unit for quantity_dispensed
            rx_to_update = self.get_prescription_details(target_rx_id_for_update)
            med_info_for_dispense = next((m for m in self.medications_catalog if m["name"] == rx_to_update["medication_name"]), {"unit":"units"})

            self.update_prescription_status(target_rx_id_for_update, "dispensed", "pharmacy001", quantity_dispensed=f"14 {med_info_for_dispense['unit']}")
            updated_rx = self.get_prescription_details(target_rx_id_for_update)
            if updated_rx:
                 print(f"  New status: {updated_rx['status']}, Dispensed Qty: {updated_rx['quantity_dispensed']}")
        else:
            print("  No prescription available to demonstrate status update.")

        print("------------------------------------")

if __name__ == '__main__':
    screen = PrescriptionManagementScreen()
    screen.display_screen()
